e2e/tests/nightly/admin/lock-workspace.spec.ts

```diff
@@ -0,0 +1,154 @@
+    expect(await aboutPage.getShareButton().isCursorNotAllowed()).toBe(true);
+    await aboutPage.verifyLockedWorkspaceActionOptions();
+    // verify the user is able to navigate to edit page to update the description
+    await aboutPage.getAboutEditIcon();
+    const workspaceEdit = new WorkspaceEditPage(page);
+    await workspaceEdit.waitForLoad();
+    // verify user is able to edit the locked workspace
+    workspaceEditedName = await workspaceEdit.fillOutRequiredDuplicationFields();
+    const updateButton = workspaceEdit.getUpdateWorkspaceButton();
+    await updateButton.waitUntilEnabled();
+    await workspaceEdit.clickCreateFinishButton(updateButton);
+    await aboutPage.waitForLoad();
+  });
+
+  test('Verify the workspace is unlocked', async () => {
+    await navigation.navMenu(page, NavLink.WORKSPACE_ADMIN);
+    const workspaceAdminPage = new WorkspaceAdminPage(page);
+    await workspaceAdminPage.waitForLoad();
+    await workspaceAdminPage.getWorkspaceNamespaceInput().type(workspaceNamespace);
+    workspaceAdminPage.clickLoadWorkspaceButton();
+    await workspaceAdminPage.waitForLoad();
+    await workspaceAdminPage.clickUnlockWorkspaceButton(workspaceStatus.Unlock);
+    await workspaceAdminPage.waitForLoad();
+    expect(workspaceAdminPage.getLockWorkspaceButton(workspaceStatus.Lock));
+    await new HomePage(page).load();
+    const workspaceCard = await WorkspaceCard.findCard(page, workspaceEditedName);
+    expect(await workspaceCard.getWorkspaceLockedIcon()).toBeFalsy();
+    await workspaceCard.verifyWorkspaceCardMenuOptions();
+    await workspaceCard.clickWorkspaceName(true);
+    const dataPage = new WorkspaceDataPage(page);
+    await dataPage.waitForLoad();
+    // verify DATA & ANALYSIS tabs are aactive and accessible
+    const analysisPage = new WorkspaceAnalysisPage(page);
+    await openTab(page, Tabs.Analysis, analysisPage);
+    const aboutPage = new WorkspaceAboutPage(page);
+    await openTab(page, Tabs.About, aboutPage);
+    await aboutPage.waitForLoad();
+    expect(await aboutPage.getShareButton().isCursorNotAllowed()).toBe(false);
+    expect(await aboutPage.getAboutLockedWorkspaceIcon()).toBeFalsy();
+    const snowmanMenu = await aboutPage.getWorkspaceActionMenu();
+    expect(await snowmanMenu.isOptionDisabled(MenuOption.Duplicate)).toBe(false);
+    expect(await snowmanMenu.isOptionDisabled(MenuOption.Edit)).toBe(false);
+    expect(await snowmanMenu.isOptionDisabled(MenuOption.Share)).toBe(false);
+    expect(await snowmanMenu.isOptionDisabled(MenuOption.Delete)).toBe(false);
+    await aboutPage.getAboutEditIcon();
+    const workspaceEdit = new WorkspaceEditPage(page);
+    await workspaceEdit.waitForLoad();
+    await workspaceEdit.clickCancelButton();
+    // cleanup
+    await dataPage.deleteWorkspace();
+  });
+
+  // function to create cohort, dataset and notebook
+  async function createDatasetNotebook(page: Page, pyNotebookName: string): Promise<NotebookPreviewPage> {
```

Comment:
This function is same as `workspace-admin-ui.spec` `createDatasetNotebook()`. I'm editing it in https://github.com/all-of-us/workbench/pull/6342. You can make this a common function in `test-utils.ts`.